src/client/datascience/baseJupyterSession.ts

```diff
@@ -112,23 +117,23 @@ export abstract class BaseJupyterSession implements IJupyterSession {
             }
             this.setSession(undefined);
             this.restartSessionPromise = undefined;
-        }
-        if (this.onStatusChangedEvent) {
+            this.onStatusChangedEvent.fire('dead');
+            this._disposed.fire();
+            this._disposed.dispose();
             this.onStatusChangedEvent.dispose();
         }
         traceInfo('Shutdown session -- complete');
     }
-    public async interrupt(timeout: number): Promise<void> {
+    public async interrupt(): Promise<void> {
```

Comment:
Removed the arguments this way the signature is identical to Jupyter Labs `Kernel.IKernelConnection.interrupt`